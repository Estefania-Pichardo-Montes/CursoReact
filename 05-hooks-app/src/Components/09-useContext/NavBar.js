import React from 'react';
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <Link to="/" className="navbar-brand">UseContext</Link>

        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
            <div className="navbar-nav">
              <NavLink activeClassName="active" to="/" className="nav-link active" >Home</NavLink>
              <NavLink activeClassName="active" to="/About" className="nav-link" >About</NavLink>
              <NavLink activeClassName="active" to="/Login" className="nav-link" >Login</NavLink>
            </div>
         </div>
        </nav>
    )
}
