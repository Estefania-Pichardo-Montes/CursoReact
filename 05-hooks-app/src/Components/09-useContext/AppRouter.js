import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";
  import {AboutScreen} from './AboutScreen';
  import {LoginScreen} from './LoginScreen';
  import {HomeScreen} from './HomeScreen';
import { NavBar } from './NavBar';


  

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <div>
                    <NavBar/>
                    <div className="container">
                     <Switch>
                         <Route exact path="/" component={HomeScreen}></Route>
                         <Route exact path="/about" component={AboutScreen }></Route>
                         <Route exact path="/login" component={LoginScreen}></Route>
                         <Redirect to="/"></Redirect>

                       </Switch>
                    </div>
                </div>
            </Router>   
        </div>
    )
}
