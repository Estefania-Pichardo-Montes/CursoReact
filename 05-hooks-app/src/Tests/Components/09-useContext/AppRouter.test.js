import React from 'react';
import {mount, shallow} from 'enzyme';
import {AppRouter} from '../../../Components/09-useContext/AppRouter';
import { UserContext } from '../../../Components/09-useContext/UserContext';

describe('Pruebas en appRouter', () => {
    const user={
        id:1,
        name:'Estefania'
    }
    const wrapper=mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter/>
        </UserContext.Provider>
    );
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();  
    })
    
})
