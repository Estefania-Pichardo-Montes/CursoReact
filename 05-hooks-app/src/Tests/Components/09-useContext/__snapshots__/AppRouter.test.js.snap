// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en appRouter Debe de mostrarse correctamente 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/About"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </nav>
    <div
      className="container"
    >
      <div>
        <h1>
          Home Screen
        </h1>
        <hr />
        <pre>
          {
   "id": 1,
   "name": "Estefania"
}
        </pre>
      </div>
    </div>
  </div>
</div>
`;
